packages/redux/src/products/selectors/grouping.ts

```diff
@@ -0,0 +1,75 @@
+import { getError, getIsLoading } from '../reducer/grouping';
+import type { ProductEntity } from '../../entities/types';
+import type { StoreState } from '../../types';
+
+/**
+ * Returns the loading grouping condition to a specific product.
+ *
+ * @param state - Application state.
+ * @param id    - Product id.
+ *
+ * @returns If the product grouping is loading or not.
+ */
+export const isProductGroupingLoading = (
+  state: StoreState,
+  id: ProductEntity['id'],
+  hash: string,
+) => {
+  const grouping = state.products?.grouping;
+  return grouping ? getIsLoading(grouping)[id]?.[hash] : undefined;
```

Comment:
You can avoid the check for the grouping property. Assume that the grouping property will always be available on the redux state. If the developer calls this method and did not provide the `grouping` reducer inside the `product` reducer, it will throw an error so the caller can properly know that it needs to configure the `grouping` reducer instead of returning `undefined`.

You can do it like this:

```suggestion
   return getIsLoading((state.products as ProductsState).grouping)[id]?.[hash];
```